contracts/placeholder/placeholder_verifier.sol

```diff
@@ -125,48 +124,35 @@ library placeholder_verifier {
                             }
                         }
                     }
-                    local_vars.variable_values_evaluation_points[i][j] = local_vars.e;
+                    local_vars.evaluation_points[i][j+1] = local_vars.e;
                 }
             unchecked{j++;}
             }
         unchecked{i++;}
         }
 
         for (uint256 i = ar_params.witness_columns; i < ar_params.witness_columns + ar_params.public_input_columns;) {
-            local_vars.variable_values_evaluation_points[i] = challenge_point;
+            local_vars.evaluation_points[i] = challenge_point;
             unchecked{i++;}
         }
         profiling.end_block();
         if (!batched_lpc_verifier.parse_verify_proof_be(blob, proof_map.eval_proof_variable_values_offset,
-            local_vars.variable_values_evaluation_points, tr_state, fri_params)) {
-            //require(false, "Wrong variable values LPC proof");
+            local_vars.evaluation_points, tr_state, fri_params)) {
+            require(false, "Wrong variable values LPC proof");
             return false;
         }
 
         profiling.end_block();
         // permutation
         profiling.start_block("PV::permutation");
-        local_vars.evaluation_points = new uint256[][](1);
-        local_vars.evaluation_points[0] = new uint256[](2);
-        assembly {
-            let addr:= mload(add(mload(add(local_vars, EVALUATION_POINTS_OFFSET)), 0x20))
-            mstore(
-                // local_vars.evaluation_points[0][1]
-                add(addr, 0x20),
-                // (local_vars.challenge * common_data.omega) % fri_params.modulus
-                mload(add(local_vars, CHALLENGE_OFFSET))
-            )
-            mstore(
-                // local_vars.evaluation_points[0][1]
-                add(addr, 0x40),
-                // (local_vars.challenge * common_data.omega) % fri_params.modulus
-                challenge_omega
-            )
-        }
+        local_vars.evaluation_points = new uint256[4][](1);
+        local_vars.evaluation_points[0][0] = 2;
+        local_vars.evaluation_points[0][1] = local_vars.challenge;
+        local_vars.evaluation_points[0][2] = challenge_omega;
 
         if (!batched_lpc_verifier.parse_verify_proof_be(blob, proof_map.eval_proof_permutation_offset,
             local_vars.evaluation_points, tr_state, fri_params)) {
-         //   require(false, "Wrong permutation LPC proof");
+            require(false, "Wrong permutation LPC proof");
```

Comment:
In my understanding, we don't need require here because we return false. It's up to the caller check if it's true or false. However, it's not clear how to return the reason of "false" in this case